File: scripts/generate_knowledge_data.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
知识长廊数据生成脚本
将 chinese-poetry 数据整合到溯光知识长廊
"""

import json
import os
from datetime import datetime

# 数据路径
POETRY_ROOT = os.path.join(os.path.dirname(__file__), '..', 'chinese-poetry-master')
BACKEND_DATA = os.path.join(os.path.dirname(__file__), '..', 'backend', 'data')
BACKEND_DB = os.path.join(os.path.dirname(__file__), '..', 'backend', 'db')

def load_json(filepath):
    """安全加载 JSON 文件"""
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            return json.load(f)
    except Exception as e:
        print(f"  读取失败: {e}")
        return None

def load_shijing():
    """读取诗经"""
    filepath = os.path.join(POETRY_ROOT, '诗经', 'shijing.json')
    data = load_json(filepath)
    if not data:
        return []

    articles = []
    for i, item in enumerate(data):
        article = {
            "id": f"shijing-{i}",
            "title": item.get('title', ''),
            "category": "诗词",
            "excerpt": (item.get('content', [''])[0] or '')[:50] + '...',
            "body": f"【{item.get('chapter', '')} · {item.get('section', '')}】\n\n" + '\n'.join(item.get('content', [])),
            "favorites": 500 + (i * 7) % 3000,
            "cover": "🪶",
            "source": "诗经",
            "author": "佚名",
            "tags": [item.get('chapter', ''), item.get('section', ''), "诗经", "国风"],
            "created_at": f"2025-01-{min(28, i+1):02d}T00:00:00Z"
        }
        articles.append(article)
    return articles

def load_lunyu():
    """读取论语"""
    filepath = os.path.join(POETRY_ROOT, '论语', 'lunyu.json')
    data = load_json(filepath)
    if not data:
        return []

    articles = []
    for i, item in enumerate(data):
        article = {
            "id": f"lunyu-{i}",
            "title": item.get('chapter', f'论语第{i+1}篇'),
            "category": "典籍",
            "excerpt": (item.get('content', '') or '')[:50] + '...',
            "body": f"【{item.get('chapter', '')}】\n\n{item.get('content', '')}",
            "favorites": 300 + (i * 11) % 2000,
            "cover": "📜",
            "source": "论语",
            "author": "孔子及其弟子",
            "tags": ["论语", "孔子", "儒家", "典籍"],
            "created_at": f"2025-02-{min(28, i+1):02d}T00:00:00Z"
        }
        articles.append(article)
    return articles

def load_chuci():
    """读取楚辞"""
    filepath = os.path.join(POETRY_ROOT, '楚辞', 'chuci.json')
    data = load_json(filepath)
    if not data:
        return []

    articles = []
    for i, item in enumerate(data):
        content = item.get('content', '') or ''
        if isinstance(content, list):
            content = '\n'.join(content)
        article = {
            "id": f"chuci-{i}",
            "title": item.get('title', ''),
            "category": "诗词",
            "excerpt": content[:50] + '...' if len(content) > 50 else content,
            "body": f"【{item.get('section', '楚辞')}】\n\n{content}",
            "favorites": 400 + (i * 13) % 2500,
            "cover": "🌿",
            "source": "楚辞",
            "author": item.get('author', '屈原'),
            "tags": ["楚辞", "屈原", item.get('title', '')],
            "created_at": f"2025-03-{min(28, i+1):02d}T00:00:00Z"
        }
        articles.append(article)
    return articles

def load_tangshi_300():
    """读取唐诗三百首"""
    filepath = os.path.join(POETRY_ROOT, '全唐诗', '唐诗三百首.json')
    data = load_json(filepath)
    if not data:
        return []

    articles = []
    for i, item in enumerate(data):
        content = item.get('content', '') or ''
        article = {
            "id": f"tangshi-300-{i}",
            "title": f"{item.get('title', '')} · {item.get('author', '佚名')}",
            "category": "诗词",
            "excerpt": content[:50] + '...' if len(content) > 50 else content,
            "body": f"【{item.get('title', '')}】\n唐 · {item.get('author', '佚名')}\n\n{content}",
            "favorites": 1000 + (i * 17) % 5000,
            "cover": "📜",
            "source": "唐诗三百首",
            "author": item.get('author', '佚名'),
            "tags": ["唐诗", "诗词", item.get('author', '佚名')],
            "created_at": f"2025-04-{min(28, i+1):02d}T00:00:00Z"
        }
        articles.append(article)
    return articles

def load_songci_300():
    """读取宋词三百首"""
    filepath = os.path.join(POETRY_ROOT, '宋词', '宋词三百首.json')
    data = load_json(filepath)
    if not data:
        return []

    articles = []
    for i, item in enumerate(data):
        content = item.get('content', '')
        if not content and item.get('paragraphs'):
            content = '\n'.join(item['paragraphs'])
        article = {
            "id": f"songci-300-{i}",
            "title": f"{item.get('title', '')} · {item.get('author', '佚名')}",
            "category": "诗词",
            "excerpt": (content or '')[:50] + '...' if len(content or '') > 50 else content,
            "body": f"【{item.get('title', '')}】\n宋 · {item.get('author', '佚名')}\n\n{content}",
            "favorites": 800 + (i * 19) % 4000,
            "cover": "🌸",
            "source": "宋词三百首",
            "author": item.get('author', '佚名'),
            "tags": ["宋词", "词", item.get('author', '佚名')],
            "created_at": f"2025-05-{min(28, i+1):02d}T00:00:00Z"
        }
        articles.append(article)
    return articles

def load_yuanqu():
    """读取元曲（取前100首）"""
    filepath = os.path.join(POETRY_ROOT, '元曲', 'yuanqu.json')
    data = load_json(filepath)
    if not data:
        return []

    articles = []
    data = data[:100]  # 取前100首
    for i, item in enumerate(data):
        content = item.get('content', '')
        if not content and item.get('paragraphs'):
            content = '\n'.join(item['paragraphs'])
        article = {
            "id": f"yuanqu-{i}",
            "title": item.get('title', f'元曲{i+1}'),
            "category": "诗词",
            "excerpt": (content or '')[:50] + '...' if len(content or '') > 50 else content,
            "body": f"【{item.get('title', '')}】\n元 · {item.get('author', '佚名')}\n\n{content}",
            "favorites": 300 + (i * 23) % 2000,
            "cover": "🎭",
            "source": "元曲",
            "author": item.get('author', '佚名'),
            "tags": ["元曲", "戏曲", item.get('author', '佚名')],
            "created_at": f"2025-06-{min(28, i+1):02d}T00:00:00Z"
        }
        articles.append(article)
    return articles

def load_youmengying():
    """读取幽梦影（取前100则）"""
    filepath = os.path.join(POETRY_ROOT, '幽梦影', 'youmengying.json')
    data = load_json(filepath)
    if not data:
        return []

    articles = []
    data = data[:100]  # 取前100则
    for i, item in enumerate(data):
        content = item.get('content', '') or item.get('text', '')
        article = {
            "id": f"youmengying-{i}",
            "title": f"幽梦影 · 第{i+1}则",
            "category": "典籍",
            "excerpt": (content or '')[:50] + '...' if len(content or '') > 50 else content,
            "body": f"【幽梦影】\n\n{content}\n\n—— 张潮",
            "favorites": 200 + (i * 29) % 1500,
            "cover": "🌙",
            "source": "幽梦影",
            "author": "张潮",
            "tags": ["幽梦影", "张潮", "明清文学"],
            "created_at": f"2025-07-{min(28, i+1):02d}T00:00:00Z"
        }
        articles.append(article)
    return articles

def load_nalan():
    """读取纳兰性德"""
    filepath = os.path.join(POETRY_ROOT, '纳兰性德', '纳兰性德诗集.json')
    data = load_json(filepath)
    if not data:
        return []

    articles = []
    for i, item in enumerate(data):
        content = item.get('content', '') or ''
        article = {
            "id": f"nalan-{i}",
            "title": item.get('title', f'纳兰词{i+1}'),
            "category": "诗词",
            "excerpt": content[:50] + '...' if len(content) > 50 else content,
            "body": f"【{item.get('title', '')}】\n清 · 纳兰性德\n\n{content}",
            "favorites": 600 + (i * 31) % 3500,
            "cover": "💧",
            "source": "纳兰性德诗集",
            "author": "纳兰性德",
            "tags": ["纳兰词", "清词", "纳兰性德"],
            "created_at": f"2025-08-{min(28, i+1):02d}T00:00:00Z"
        }
        articles.append(article)
    return articles

def load_sishu_wujing():
    """读取四书五经"""
    articles = []

    # 大学
    filepath = os.path.join(POETRY_ROOT, '四书五经', 'daxue.json')
    data = load_json(filepath)
    if data:
        for i, item in enumerate(data):
            content = item.get('content', '') or ''
            article = {
                "id": f"daxue-{i}",
                "title": item.get('chapter', '大学'),
                "category": "典籍",
                "excerpt": content[:50] + '...' if len(content) > 50 else content,
                "body": f"【大学】\n\n{content}",
                "favorites": 800 + i * 50,
                "cover": "📜",
                "source": "四书五经",
                "author": "曾子",
                "tags": ["四书", "大学", "儒家"],
                "created_at": f"2025-09-{min(28, i+1):02d}T00:00:00Z"
            }
            articles.append(article)

    # 中庸
    filepath = os.path.join(POETRY_ROOT, '四书五经', 'zhongyong.json')
    data = load_json(filepath)
    if data:
        for i, item in enumerate(data):
            content = item.get('content', '') or ''
            article = {
                "id": f"zhongyong-{i}",
                "title": item.get('chapter', '中庸'),
                "category": "典籍",
                "excerpt": content[:50] + '...' if len(content) > 50 else content,
                "body": f"【中庸】\n\n{content}",
                "favorites": 750 + i * 45,
                "cover": "📜",
                "source": "四书五经",
                "author": "子思",
                "tags": ["四书", "中庸", "儒家"],
                "created_at": f"2025-09-{min(28, i+50):02d}T00:00:00Z"
            }
            articles.append(article)

    # 孟子
    filepath = os.path.join(POETRY_ROOT, '四书五经', 'mengzi.json')
    data = load_json(filepath)
    if data:
        for i, item in enumerate(data):
            content = item.get('content', '') or ''
            article = {
                "id": f"mengzi-{i}",
                "title": item.get('chapter', '孟子'),
                "category": "典籍",
                "excerpt": content[:50] + '...' if len(content) > 50 else content,
                "body": f"【{item.get('chapter', '孟子')}】\n\n{content}",
                "favorites": 700 + i * 40,
                "cover": "📜",
                "source": "四书五经",
                "author": "孟子及其弟子",
                "tags": ["四书", "孟子", "儒家"],
                "created_at": f"2025-10-{min(28, i+1):02d}T00:00:00Z"
            }
            articles.append(article)

    return articles

def main():
    """主函数"""
    print("=" * 60)
    print("溯光知识长廊数据生成脚本")
    print("=" * 60)
    print()

    articles = []

    # 加载各类数据
    loaders = [
        ("诗经", load_shijing),
        ("论语", load_lunyu),
        ("楚辞", load_chuci),
        ("唐诗三百首", load_tangshi_300),
        ("宋词三百首", load_songci_300),
        ("元曲", load_yuanqu),
        ("幽梦影", load_youmengying),
        ("纳兰性德", load_nalan),
        ("四书五经", load_sishu_wujing),
    ]

    for name, loader in loaders:
        print(f"正在读取 {name}...")
        try:
            data = loader()
            if data:
                articles.extend(data)
                print(f"  ✓ 成功读取 {len(data)} 篇")
            else:
                print(f"  ✗ 读取失败或数据为空")
        except Exception as e:
            print(f"  ✗ 错误: {e}")

    # 读取现有的知识文章
    existing_file = os.path.join(BACKEND_DATA, 'knowledge_articles.json')
    if os.path.exists(existing_file):
        print(f"\n正在读取现有知识文章...")
        with open(existing_file, 'r', encoding='utf-8') as f:
            existing = json.load(f)
        print(f"  ✓ 成功读取 {len(existing)} 篇")
        articles.extend(existing)

    # 去重
    seen_ids = set()
    unique_articles = []
    for item in articles:
        if item['id'] not in seen_ids:
            seen_ids.add(item['id'])
            unique_articles.append(item)

    print(f"\n总计: {len(unique_articles)} 篇文章")
    print()

    # 保存完整数据
    os.makedirs(BACKEND_DATA, exist_ok=True)
    output_file = os.path.join(BACKEND_DATA, 'knowledge_articles_full.json')
    with open(output_file, 'w', encoding='utf-8') as f:
        json.dump(unique_articles, f, ensure_ascii=False, indent=2)
    print(f"✓ 已保存到: {output_file}")

    # 生成 SQL 导入脚本
    os.makedirs(BACKEND_DB, exist_ok=True)
    sql_file = os.path.join(BACKEND_DB, 'seed_poetry_data.sql')

    with open(sql_file, 'w', encoding='utf-8') as f:
        f.write(f"-- 诗词典籍数据导入\n")
        f.write(f"-- 生成时间: {datetime.now().isoformat()}\n")
        f.write(f"-- 总计: {len(unique_articles)} 篇文章\n\n")

        for item in unique_articles:
            tags = item.get('tags', [])
            tags_str = '{' + ','.join(f'"{t}"' for t in tags) + '}'

            body = (item.get('body') or '').replace("'", "''").replace("\n", "\\n")
            excerpt = (item.get('excerpt') or '').replace("'", "''")
            title = (item.get('title') or '').replace("'", "''")

            f.write(f"""INSERT INTO knowledge_articles (id, title, category, excerpt, body, cover, source, author, tags, favorites, created_at)
VALUES (
  '{item['id']}',
  E'{title}',
  '{item.get('category', '诗词')}',
  E'{excerpt}',
  E'{body}',
  '{item.get('cover', '📜')}',
  '{item.get('source', '')}',
  '{item.get('author', '匿名')}',
  ARRAY{tags_str}::text[],
  {item.get('favorites', 0)},
  '{item.get('created_at', datetime.now().isoformat())}'
) ON CONFLICT (id) DO NOTHING;

""")

    print(f"✓ SQL脚本已保存到: {sql_file}")

    # 生成前端数据
    frontend_data = []
    for item in unique_articles[:200]:  # 前端只保留200篇
        frontend_data.append({
            "id": item['id'],
            "title": item['title'],
            "category": item['category'],
            "excerpt": item['excerpt'],
            "content": item['body'],
            "favorites": item['favorites'],
            "cover": item['cover']
        })

    frontend_file = os.path.join(os.path.dirname(__file__), '..', 'src', 'lib', 'knowledge-data-full.ts')
    os.makedirs(os.path.dirname(frontend_file), exist_ok=True)

    with open(frontend_file, 'w', encoding='utf-8') as f:
        f.write("// 知识长廊完整数据\n")
        f.write(f"// 共 {len(frontend_data)} 篇文章\n\n")
        f.write("export interface Article {\n")
        f.write("  id: string;\n")
        f.write("  title: string;\n")
        f.write("  category: string;\n")
        f.write("  excerpt: string;\n")
        f.write("  content: string;\n")
        f.write("  favorites: number;\n")
        f.write("  cover: string;\n")
        f.write("}\n\n")
        f.write(f"export const ARTICLES_FULL: Article[] = {json.dumps(frontend_data, ensure_ascii=False, indent=2)} as const;\n")

    print(f"✓ 前端数据已保存到: {frontend_file}")

    print()
    print("=" * 60)
    print("生成完成！")
    print("=" * 60)

if __name__ == '__main__':
    main()

File: scripts/generate-knowledge-data.ts
import fs from 'fs';
import path from 'path';

// 数据路径
const POETRY_ROOT = path.join(process.cwd(), '../chinese-poetry-master');

// 读取诗经
function loadShijing() {
  const data = JSON.parse(fs.readFileSync(path.join(POETRY_ROOT, '诗经/shijing.json'), 'utf-8'));
  return data.map((item, index) => ({
    id: `shijing-${index}`,
    title: item.title,
    category: '诗词',
    excerpt: item.content[0].substring(0, 50) + '...',
    body: `【${item.chapter} · ${item.section}】\n\n${item.content.join('\n')}`,
    favorites: Math.floor(Math.random() * 3000) + 500,
    cover: '🪶',
    source: '诗经',
    author: '佚名',
    tags: [item.chapter, item.section, '诗经', '国风'],
    created_at: new Date(2025, 0, 1 + index).toISOString()
  }));
}

// 读取论语
function loadLunyu() {
  const data = JSON.parse(fs.readFileSync(path.join(POETRY_ROOT, '论语/lunyu.json'), 'utf-8'));
  return data.map((item, index) => ({
    id: `lunyu-${index}`,
    title: item.chapter,
    category: '典籍',
    excerpt: item.content.substring(0, 50) + '...',
    body: `【${item.chapter}】\n\n${item.content}`,
    favorites: Math.floor(Math.random() * 2000) + 300,
    cover: '📜',
    source: '论语',
    author: '孔子及其弟子',
    tags: ['论语', '孔子', '儒家', '典籍'],
    created_at: new Date(2025, 1, 1 + index).toISOString()
  }));
}

// 读取楚辞
function loadChuci() {
  const data = JSON.parse(fs.readFileSync(path.join(POETRY_ROOT, '楚辞/chuci.json'), 'utf-8'));
  return data.map((item, index) => ({
    id: `chuci-${index}`,
    title: item.title,
    category: '诗词',
    excerpt: item.content.substring(0, 50) + '...',
    body: `【${item.section || '楚辞'}】\n\n${item.content}`,
    favorites: Math.floor(Math.random() * 2500) + 400,
    cover: '🌿',
    source: '楚辞',
    author: item.author || '屈原',
    tags: ['楚辞', '屈原', '离骚'],
    created_at: new Date(2025, 2, 1 + index).toISOString()
  }));
}

// 读取唐诗三百首
function loadTangShi300() {
  const data = JSON.parse(fs.readFileSync(path.join(POETRY_ROOT, '全唐诗/唐诗三百首.json'), 'utf-8'));
  return data.map((item, index) => ({
    id: `tangshi-300-${index}`,
    title: `${item.title} · ${item.author}`,
    category: '诗词',
    excerpt: item.content.substring(0, 50) + '...',
    body: `【${item.title}】\n唐 · ${item.author}\n\n${item.content}\n\n【注释】${item.comment || '暂无注释'}\n【译文】${item.translate || '暂无译文'}`,
    favorites: Math.floor(Math.random() * 5000) + 1000,
    cover: '📜',
    source: '唐诗三百首',
    author: item.author,
    tags: ['唐诗', '诗词', item.author],
    created_at: new Date(2025, 3, 1 + index).toISOString()
  }));
}

// 读取宋词三百首
function loadSongCi300() {
  const data = JSON.parse(fs.readFileSync(path.join(POETRY_ROOT, '宋词/宋词三百首.json'), 'utf-8'));
  return data.map((item, index) => ({
    id: `songci-300-${index}`,
    title: `${item.title} · ${item.author}`,
    category: '诗词',
    excerpt: (item.content || item.paragraphs?.join('，'))?.substring(0, 50) + '...',
    body: `【${item.title}】\n宋 · ${item.author}\n\n${item.content || item.paragraphs?.join('\n')}\n\n【词牌】${item.rhythnic || '未知'}`,
    favorites: Math.floor(Math.random() * 4000) + 800,
    cover: '🌸',
    source: '宋词三百首',
    author: item.author,
    tags: ['宋词', '词', item.author],
    created_at: new Date(2025, 4, 1 + index).toISOString()
  }));
}

// 读取元曲
function loadYuanQu() {
  const data = JSON.parse(fs.readFileSync(path.join(POETRY_ROOT, '元曲/yuanqu.json'), 'utf-8'));
  return data.slice(0, 100).map((item, index) => ({
    id: `yuanqu-${index}`,
    title: item.title,
    category: '诗词',
    excerpt: item.content?.substring(0, 50) || item.paragraphs?.join('，')?.substring(0, 50) + '...',
    body: `【${item.title}】\n元 · ${item.author || '佚名'}\n\n${item.content || item.paragraphs?.join('\n')}`,
    favorites: Math.floor(Math.random() * 2000) + 300,
    cover: '🎭',
    source: '元曲',
    author: item.author || '佚名',
    tags: ['元曲', '戏曲', item.author],
    created_at: new Date(2025, 5, 1 + index).toISOString()
  }));
}

// 读取幽梦影
function loadYouMengYing() {
  const data = JSON.parse(fs.readFileSync(path.join(POETRY_ROOT, '幽梦影/youmengying.json'), 'utf-8'));
  return data.slice(0, 100).map((item, index) => ({
    id: `youmengying-${index}`,
    title: item.title || `幽梦影·第${index + 1}则`,
    category: '典籍',
    excerpt: item.content?.substring(0, 50) || '暂无内容...',
    body: `【幽梦影】\n\n${item.content || item.text || ''}\n\n—— 张潮`,
    favorites: Math.floor(Math.random() * 1500) + 200,
    cover: '🌙',
    source: '幽梦影',
    author: '张潮',
    tags: ['幽梦影', '张潮', '明清文学'],
    created_at: new Date(2025, 6, 1 + index).toISOString()
  }));
}

// 读取纳兰性德
function loadNalan() {
  const data = JSON.parse(fs.readFileSync(path.join(POETRY_ROOT, '纳兰性德/纳兰性德诗集.json'), 'utf-8'));
  return data.map((item, index) => ({
    id: `nalan-${index}`,
    title: item.title,
    category: '诗词',
    excerpt: item.content?.substring(0, 50) || '暂无内容...',
    body: `【${item.title}】\n清 · 纳兰性德\n\n${item.content || ''}`,
    favorites: Math.floor(Math.random() * 3500) + 600,
    cover: '💧',
    source: '纳兰性德诗集',
    author: '纳兰性德',
    tags: ['纳兰词', '清词', '纳兰性德'],
    created_at: new Date(2025, 7, 1 + index).toISOString()
  }));
}

// 主函数：整合所有数据
async function generateKnowledgeData() {
  console.log('开始生成知识长廊数据...');

  const articles = [];

  try {
    console.log('读取诗经...');
    articles.push(...loadShijing());
    console.log(`诗经: ${loadShijing().length} 篇`);
  } catch (e) {
    console.log('诗经读取失败:', e.message);
  }

  try {
    console.log('读取论语...');
    articles.push(...loadLunyu());
    console.log(`论语: ${loadLunyu().length} 篇`);
  } catch (e) {
    console.log('论语读取失败:', e.message);
  }

  try {
    console.log('读取楚辞...');
    articles.push(...loadChuci());
    console.log(`楚辞: ${loadChuci().length} 篇`);
  } catch (e) {
    console.log('楚辞读取失败:', e.message);
  }

  try {
    console.log('读取唐诗三百首...');
    articles.push(...loadTangShi300());
    console.log(`唐诗三百首: ${loadTangShi300().length} 篇`);
  } catch (e) {
    console.log('唐诗三百首读取失败:', e.message);
  }

  try {
    console.log('读取宋词三百首...');
    articles.push(...loadSongCi300());
    console.log(`宋词三百首: ${loadSongCi300().length} 篇`);
  } catch (e) {
    console.log('宋词三百首读取失败:', e.message);
  }

  try {
    console.log('读取元曲...');
    articles.push(...loadYuanQu());
    console.log(`元曲: ${loadYuanQu().length} 篇`);
  } catch (e) {
    console.log('元曲读取失败:', e.message);
  }

  try {
    console.log('读取幽梦影...');
    articles.push(...loadYouMengYing());
    console.log(`幽梦影: ${loadYouMengYing().length} 篇`);
  } catch (e) {
    console.log('幽梦影读取失败:', e.message);
  }

  try {
    console.log('读取纳兰性德...');
    articles.push(...loadNalan());
    console.log(`纳兰性德: ${loadNalan().length} 篇`);
  } catch (e) {
    console.log('纳兰性德读取失败:', e.message);
  }

  // 读取现有的知识文章
  const existingFile = path.join(process.cwd(), '../backend/data/knowledge_articles.json');
  if (fs.existsSync(existingFile)) {
    const existing = JSON.parse(fs.readFileSync(existingFile, 'utf-8'));
    console.log(`现有知识文章: ${existing.length} 篇`);
    // 合并现有数据
    articles.push(...existing);
  }

  // 去重
  const uniqueArticles = articles.filter((item, index, self) =>
    index === self.findIndex(t => t.id === item.id)
  );

  console.log(`\n总计: ${uniqueArticles.length} 篇文章`);

  // 保存到文件
  const outputPath = path.join(process.cwd(), '../backend/data/knowledge_articles_full.json');
  fs.writeFileSync(outputPath, JSON.stringify(uniqueArticles, null, 2), 'utf-8');
  console.log(`已保存到: ${outputPath}`);

  // 生成 SQL 导入脚本
  const sqlPath = path.join(process.cwd(), '../backend/db/seed_poetry_data.sql');
  const sqlStatements = uniqueArticles.map(item => {
    const tags = item.tags || [];
    return `INSERT INTO knowledge_articles (id, title, category, excerpt, body, cover, source, author, tags, favorites, created_at)
VALUES (
  '${item.id}',
  E'${(item.title || '').replace(/'/g, "''")}',
  '${item.category}',
  E'${(item.excerpt || '').replace(/'/g, "''")}',
  E'${(item.body || '').replace(/'/g, "''")}',
  '${item.cover || '📜'}',
  '${item.source || ''}',
  '${item.author || '匿名'}',
  ARRAY[${tags.map(t => `'${t}'`).join(', ')}],
  ${item.favorites || 0},
  '${item.created_at || new Date().toISOString()}'
) ON CONFLICT (id) DO NOTHING;`;
  }).join('\n\n');

  fs.writeFileSync(sqlPath, `-- 诗词典籍数据导入\n-- 生成时间: ${new Date().toISOString()}\n-- 总计: ${uniqueArticles.length} 篇文章\n\n${sqlStatements}`, 'utf-8');
  console.log(`SQL脚本已保存到: ${sqlPath}`);

  return uniqueArticles;
}

generateKnowledgeData().catch(console.error);
